### PR TITLE
feat: update built-in readme

### PR DESCRIPTION
**Pull request by LucasXu0**
Update built-in README for new users.

- [x] Getting started (document)
  - [x] Desktop guide (document)
  - [x] Mobile guide (document)
- [x] To-Dos (board)
  ~~- [ ] Inbox (grid)~~
~~- [ ] Ask AI (ai chat)~~
- [x] add tests

https://github.com/AppFlowy-IO/AppFlowy-Collab/pull/264

### CODE REVIEW

**Comment by appflowy** [comment]
can we generate by code Instead of JSON file?

File: src/biz/user/user_init.rs [hunk 1]
```diff
@@ -130,14 +147,23 @@ async fn create_workspace_database_collab(
...
   {
     let mut txn = collab.context.transact_mut();
-    let _ = collab
+    let workspace_databases = collab
       .data
       .get_or_init::<_, ArrayRef>(&mut txn, WORKSPACE_DATABASES);
+    // insert the initial database records
+    for (object_id, database_id) in initial_database_records {
+      let map_ref = workspace_databases.push_back(&mut txn, MapPrelim::default());
+      // TODO(Lucas): use the const key here.
```
Comment (LucasXu0) [inline src/biz/user/user_init.rs:161]: These values are defined in the collab-database. Should we import the collab-database as a dependency here?

File: libs/workspace-template/src/document/getting_started.rs [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+
+    // 2. create a new document with the getting started data
+    let data = tokio::task::spawn_blocking(move || {
+      let collab = Collab::new_with_origin(CollabOrigin::Empty, &object_id, vec![], false);
+      let document = Document::open_with(collab, Some(document_data))?;
+      let data = document.encode_collab()?;
+
+      Ok::<_, anyhow::Error>(TemplateData {
+        object_id,
+        object_type: CollabType::Document,
```
Comment (appflowy) [inline libs/workspace-template/src/document/getting_started.rs:54]: how about rename object_type to collab_type. object_data to encode_collab

File: libs/workspace-template/src/document/getting_started.rs [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+          }
+        });
+
+    let mut template_data = vec![database_template_data];
+    template_data.extend(database_row_template_data);
+
+    Ok(template_data)
+  }
+
+  pub async fn create_document_and_database_data(
```
Comment (appflowy) [inline libs/workspace-template/src/document/getting_started.rs:118]: would it be private? if yes, remove `pub`

File: libs/workspace-template/src/lib.rs [hunk 4]
```diff
@@ -14,26 +14,34 @@ use tokio::sync::RwLock;
...
 }
 
 pub struct TemplateData {
   pub object_id: String,
   pub object_type: CollabType,
   pub object_data: EncodedCollab,
+
+  // only for the database template
+  // it used to reference the database id from object_id
+  pub database_id: Option<String>,
```
Comment (appflowy) [inline libs/workspace-template/src/lib.rs:44]: how about using `enum` instead of using database_id?

File: libs/workspace-template/src/lib.rs [hunk 5]
```diff
@@ -14,26 +14,41 @@ use tokio::sync::RwLock;
...
+  ) -> Result<Vec<TemplateData>>;
+}
+
+#[derive(Clone)]
+pub enum TemplateObjectId {
+  Folder(String),
+  Document(String),
+  Database {
+    object_id: String,
+    // it used to reference the database id from object_id
```
Comment (appflowy) [inline libs/workspace-template/src/lib.rs:43]: it should be `It is used to reference `

**Pull request closed without merging**